### PR TITLE
ipc: Extract pipeline id dereference from ipc_comp_dev structure

### PR DESCRIPTION
**Pull request by ktrzcinx**
Mechanism of comparison pipe_id from ipc_comp_dev will be used in
trace filtering, so this part should be moved to separate function
and allow code reusability.
Moreover function usage is compact and descriptive method to define
what given block of code should do, so after such a refactor code
readability is improved.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

This is part of https://github.com/thesofproject/sof/pull/2965

### CODE REVIEW

File: src/include/sof/drivers/ipc.h [hunk 1]
```diff
@@ -124,6 +124,19 @@ static inline uint64_t ipc_task_deadline(void *data)
...
+{
+	switch (icd->type) {
+	case COMP_TYPE_COMPONENT:
+		return dev_comp_pipe_id(icd->cd);
+	case COMP_TYPE_BUFFER:
+		return icd->cb->pipeline_id;
+	case COMP_TYPE_PIPELINE:
+		return icd->pipeline->ipc_pipe.pipeline_id;
+	}
+	return -1;
```
Comment (lgirdwood) [inline src/include/sof/drivers/ipc.h:136]: default trace error and return -EINVAL

**Comment by ktrzcinx** [comment]
To implement function like this:

```
/* update `*`, `name*`, `nameX.*` or `nameX.Y` */
static int trace_filter_update_instances(int32_t uuid_id, int32_t pipe_id,
					 int32_t comp_id, int32_t log_level)
{
	struct ipc *ipc = ipc_get();
	struct ipc_comp_dev *icd;
	struct list_item *clist;
	struct tr_ctx *ctx;
	bool correct_comp;
	int cnt = 0;

	list_for_item(clist, &ipc->comp_list) {
		icd = container_of(clist, struct ipc_comp_dev, list);
		ctx = trace_filter_ipc_comp_context(icd);
		correct_comp = comp_id == -1 || icd->id == comp_id;
		correct_comp &= uuid_id == -1 || ctx->uuid_p == uuid_id;
		correct_comp &= pipe_id == -1 ||
				ipc_comp_pipe_id(icd) == pipe_id;
		if (correct_comp) {
			ctx->level = log_level;
			++cnt;
		}

		platform_shared_commit(icd, sizeof(*icd));
	}
	return cnt;
}
```

File: src/include/sof/drivers/ipc.h (hunk 1 above)
Comment (lgirdwood) [reply src/include/sof/drivers/ipc.h]: ping

Comment (ktrzcinx) [reply src/include/sof/drivers/ipc.h]: changed

File: src/include/sof/drivers/ipc.h [hunk 2]
```diff
@@ -126,7 +125,7 @@ static inline uint64_t ipc_task_deadline(void *data)
 
 static inline int32_t ipc_comp_pipe_id(const struct ipc_comp_dev *icd)
 {
-	extern tr_ctx ipc_tr;
+	extern struct tr_ctx ipc_tr;
```
Comment (lgirdwood) [inline src/include/sof/drivers/ipc.h:128]: Btw, is there any plan to move this into SOF context ? fyi @mmaka1 @tlauda

Comment (ktrzcinx) [reply src/include/sof/drivers/ipc.h:128]: @lgirdwood I'm not sure what you are asking for.

Comment (paulstelian97) [reply src/include/sof/drivers/ipc.h:128]: Given that these variables are in custom sections, I don't think they themselves can be in the SOF context (but maybe you _could_ have pointers?)

**Comment by jajanusz** [comment]
@ktrzcinx  `Internal Intel CI System/merge/build — Unit tests failed`

**Comment by ktrzcinx** [comment]
@jajanusz thanks, now looks good

**Pull request merged**